src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -2037,19 +2037,7 @@ class PartitionClauseSegment(ansi.PartitionClauseSegment):
     match_grammar = Sequence(
         "PARTITION",
         "BY",
-        Delimited(
-            OptionallyBracketed(
-                OneOf(
-                    Ref("ColumnReferenceSegment"),
-                    Bracketed(
-                        Ref("SelectStatementSegment"),
-                    ),
-                    Ref("FunctionSegment"),
-                    Ref("VariableIdentifierSegment"),
-                    "NULL",
-                ),
-            ),
-        ),
+        Delimited(OptionallyBracketed(Ref("ExpressionSegment"))),
```

Comment:
I'm tempted to go for something like `OneOf(Ref("ColumnReferenceSegment"), Ref("ExpressionSegment"))` as a hybrid of the old and the new. I imagine, often this will only be a single column, and it seems a bit far to complicate the parse tree with expressions when it's not required - but for everything else (functions, select statements etc...) then I think reaching for just `expression` is the right option.

What do you think?

Reply:
Fair enough, I've been approaching the grammar very much from a theoretical perspective but it seems to be useful to have this simplification in practice. Will change this!